crates/bevy_transform/src/hierarchy/hierarchy_maintenance_system.rs

```diff
@@ -50,11 +50,9 @@ pub fn parent_update_system(
         // `children_additions`).
         if let Ok(mut new_parent_children) = children_query.get_mut(parent.0) {
             // This is the parent
-            debug_assert!(
-                !(*new_parent_children).0.contains(&entity),
-                "children already added"
-            );
-            (*new_parent_children).0.push(entity);
+            if !(*new_parent_children).0.contains(&entity) {
```

Comment:
This seems like a reasonable short-term hack to ensure correctness in light of the suboptimal implementation we have today, but can we add a `PERF` note here to call out that in an ideal world wouldn't need to check for duplicates on insertion?

Reply:
added 👍